src/canvas-extractor/edfi_canvas_extractor/graphql/canvas.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import arrow
+import pytz
+
+
+class CanvasObject(object):
+    """
+    Base class for all classes representing objects returned by the API.
+
+    This makes a call to :func:`canvasapi.canvas_object.CanvasObject.set_attributes`
+    to dynamically construct this object's attributes with a JSON object.
+    """
+
+    def __getattribute__(self, name):
+        return super(CanvasObject, self).__getattribute__(name)
+
+    def __init__(self, requester, attributes):
+        """
+        :param requester: The requester to pass HTTP requests through.
+        :type requester: :class:`canvasapi.requester.Requester`
+        :param attributes: The JSON object to build this object with.
+        :type attributes: dict
+        """
+        self._requester = requester
+        self.set_attributes(attributes)
+
+    def __repr__(self):  # pragma: no cover
+        classname = self.__class__.__name__
+        attrs = ", ".join(
+            [
+                "{}={}".format(attr, val)
+                for attr, val in self.__dict__.items()
+                if attr != "attributes"
+            ]
+        )  # noqa
+        return "{}({})".format(classname, attrs)
```

Comment:
As a mild style preference, we've been using string interpolation almost everywhere. There was some situation with exceptions where we chose not to use it, though I can't remember the detail of why now.
```suggestion
        return f"{classname}({attrs})"
```